frontend/vue/components/Syllabus/SyllabusFormCourseInfo.vue

```diff
@@ -98,37 +105,56 @@ import 'carbon-web-components/es/components/tabs/tabs.js'
 import 'carbon-web-components/es/components/tabs/tab.js'
 import 'carbon-web-components/es/components/input/input.js'
 import BasicLink from '../common/BasicLink.vue'
+import BXInput from 'carbon-web-components/es/components/input/input.js'
+import { Syllabus } from '../../../ts/syllabus'
 
 export default defineComponent({
   name: 'SyllabusFormCourseInfo',
   components: {
     BasicLink
   },
+  props: {
+    syllabus: {
+      type: Object,
+      default: () => ({} as Syllabus),
+      required: false
+    }
+  },
   data () {
     return {
       defaultActiveTab: 'write',
       saveSyllabusInfoLink: {
         label: 'Save to syllabus',
         url: '#'
       },
-      syllabusSaved: false,
-      courseName: '',
-      courseInstructor: '',
-      courseLocation: '',
-      courseUniversity: '',
-      courseOfficeHours: '',
-      courseHours: '',
-      courseEmail: ''
+      syllabusSaved: false
     }
   },
   methods: {
     saveInfoAction () {
       this.saveSyllabusInfoLink.label = 'Saved'
       this.syllabusSaved = true
     },
-    updateFormInfo () {
+    updateFormInfo (evt: InputEvent) {
       this.saveSyllabusInfoLink.label = 'Save to syllabus'
       this.syllabusSaved = false
+
+      const updatedData = {
+        name: (this.$refs.nameInput as BXInput).value,
+        instructor: (this.$refs.instructorInput as BXInput).value,
+        location: (this.$refs.locationInput as BXInput).value,
+        institution: (this.$refs.institutionInput as BXInput).value,
+        officeHours: (this.$refs.officeHoursInput as BXInput).value,
+        classHours: (this.$refs.classHoursInput as BXInput).value,
+        email: (this.$refs.emailInput as BXInput).value
+      }
+
+      const mergedObject = {
```

Comment:
I don't know if this info can simplify your logic but for the update in backend is more than enough that you send me the fields that you want to modify. You don't need to pass the entire syllabus (just to mention).

Reply:
Per offline conversations. This is resolved :)